Software/Development/SemesterProject/App/PIController.c

```diff
@@ -62,6 +62,14 @@ float PiCalculation(PIobject *PIcontroller, float reference, float measuredValue
         // Calculating the controller output
         PIoutput = KP * error + KI * PIcontroller->integrationOfError;
 
+    if (PIcontroller->antiWindupFlag == 0)
+    {
+        if (isOutputSaturatedPositive(PIoutput))
+            PIoutput = CURRENT_LIMIT;

```

Comment:
I think that the output value of the current PIs is a voltage reference. If that is the case, maybe the output should be the maximum voltage reference. 

Also, this PI will be used for speed regulation if I'm not wrong, then maybe it would be better to add the limit to the PIController object, as another attribute. This way Speed regulation and current regulation will be able to have different maximum values.